eosclubhouse/libquest.py

```diff
@@ -532,7 +532,10 @@ async def wait_or_timeout(futures, timeout=None):
             pending = []
 
             try:
+                # The "loop" parameter is deprecated and should be removed for python 3.10
+                # https://docs.python.org/3.9/library/asyncio-task.html#waiting-primitives
                 _done, pending = await asyncio.wait(futures, timeout=timeout,
+                                                    loop=asyncio.get_event_loop(),
```

Comment:
Unfortunately I don't know anything at all about asycio, maybe someone else in the organisation does?

I guess this workaround works for now but it would be nice to understand exactly what's going on. I don't quite understand why this code creates a new event loop each time – I would have expected that asyncio had been somehow wired up to the GLib event loop.

Reply:
I don't know, in the code there are several places where a new event loop is created. I was looking at the code and this is the simplest way that I found to make it work again, without touching a lot of code.

Reply:
There are recent changes in upstream `glibcoro`, this one in particular looks promising: https://github.com/ldo/glibcoro/commit/1d2d86e6e2b33c51427c4c41ba95e059a70d5556

I don't remember why are we forking glibcoro, but is not the first time we have to backport a commit. Last time it was for a API change for Python 3.7: https://github.com/endlessm/glibcoro/pull/1

Longer term we wanted to migrate to `asyncio-glib` https://github.com/jhenstridge/asyncio-glib .